webapp-django/crashstats/crashstats/views.py

```diff
@@ -788,13 +789,16 @@ def crashes_per_day(request, default_context=None):
         supersearch_params['platform'].append('Mac')
         supersearch_params['platform'].remove('Mac OS X')
 
-    graph_data, results, adi_by_version = _get_crashes_per_day_with_adu(
-        supersearch_params,
-        start_date,
-        end_date,
-        platforms,
-        _date_range_type
-    )
+    try:
+        graph_data, results, adi_by_version = _get_crashes_per_day_with_adu(
+            supersearch_params,
+            start_date,
+            end_date,
+            platforms,
+            _date_range_type
+        )
+    except BadArgumentError as exception:
+        return http.HttpResponseBadRequest(unicode(exception))
```

Comment:
Ouch--I wouldn't have fixed #3612 that way, either. If you read through the Django code, it converts everything down to str. By passing in a unicode here, you're just passing the buck and who knows what happens in Django-land.

I think it's a mistake. If you really think that calling unicode is the right thing here, then you should do:

```
return http.HttpResponseBadRequest(unicode(exception).encode('utf-8'))
```